pyhik/hikvision.py

```diff
@@ -132,6 +134,72 @@ def current_event_states(self):
         """Return Event states dictionary"""
         return self.event_states
 
+    def get_motion_detection(self):
```

Comment:
You can leave this public, but i'd consider the action of checking the current status something that would mostly be called internally.  To adopt as core functionality I think it makes sense to add it to the initialize call so that state is determined right away.  Change it to update a variable, rather than return directly.  A property can be defined to return the variable so I/O isn't done every time we want to see what the state is.

The set function already does an update before making a change so we should always be correct internally.

Reply:
I like having it public so it can be integrated with HomeAssistant, for example.
I'll add the variable to the class and update it from here.

Regarding not doing I/O, I'd like a way to force making the call because someone may change this from the camera UI, and then it'll get out of sync.

Reply:
Integration with HASS is the driver behind most of my suggestions. :) We want to keep I/O out of the event loop.  Saving the state to a variable and defining a property will prevent that.